aws-robotics/aws-robomaker-sample-application-deepracer
# Ros2 dashing

@AAlon , Please find first PR for Deep racer application.

*Description of changes:*
**Deepracer simulation migration from ROS1 to ROS2**
- Modified CMakeLists.txt and package.xml as per ROS2 guidelines.
- Added ROS2 Deep Racer Simulation package dependencies repos to .rosinstall file.
- Added launch file to launch racer car and car track model in gazebo.
- converted .xacro file of car model to sdf file.

**Steps to build and test simulation workspace locally:**
- First add COLCON_IGNORE empty file at following paths to avoid build error as these packages are not part of this PR.
  - simulation_ws/src/deepracer_interpreter/COLCON_IGNORE
  - simulation_ws/src/deepracer_msgs/COLCON_IGNORE
  - simulation_ws/src/sagemaker_rl_agent/COLCON_IGNORE
- cd simulation_ws
- rosws update
- rosdep install --from-paths src --ignore-src -r -y
- colcon build
- export GAZEBO_MODEL_PATH=$GAZEBO_MODEL_PATH:<GAZEBO_MODELS_PATH_HERE>
   eg. export GAZEBO_MODEL_PATH=$GAZEBO_MODEL_PATH:/home/root-pc/vandana/ros2/deep_exm_ws/src/deepracer_simulation/models
- export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu/gazebo-9/plugins/
- ros2 launch deepracer_simulation racecar_simulation.launch.py

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
